NEW_README.md

```diff
@@ -141,15 +140,155 @@ Run prints its output if either,
 2. no tasks are "long" (eg a one-shot "build" procedure, rather than an ongoing
    "dev server").
 
+## Task Configuration Details
+
+The core of Run's configuration is the `tasks.toml` file, which defines the tasks you want Run to manage. Each task is specified with a range of properties, as shown in the example below:
+
+```toml
+[[task]]
+  id = "dev"
+  type = "long"
+  dependencies = ["simulate-coding"]
+  triggers = ["build-css", "build-js"]
+  watch = ["server-config.json"]
+  env = {
+    KEY = "some value"
+    OTHER_KEY = "some other value"
+  }
+  cmd = """
+    echo "dev-server running at http://localhost:3000"
+    while true; do sleep 1; done
+  """
+```
+
+Explore a [realistic project configuration](https://github.com/amonks/run/tree/main/examples/demo) to see `tasks.toml` in action.
+
+### Understanding Task Fields
+
+#### Required Fields
+
+Task definitions have two required fields: `id` and `type`. _Note_: if only the required fields are set and none of `dependencies`, `triggers`, or `cmd` are specified, then the task is a no-op.
+
+##### `id`
+
+The unique identifier for a task. It is used for:
+- Command line invocation (`$ run <id>`)
+- Identifying the task in the interactive UI (TUI).
+
+##### `type`
+
+Specifies the task's lifecycle management strategy:
+- A task can be of only one type: `long` or `short`, any other value is invalid
+- **`long`**: The task is kept alive indefinitely. It's restarted if it exits unexpectedly. **Not suitable as a trigger**.
+  - For example, a development server or a test runner.
+  - If a task depends on a "long" task, Run doesn't really know when the long task has produced whatever output is depended on, so the dependent is run 500ms after the long task starts.
```

Comment:
great detail to include 🎉 

adding the shorter configuration section above gives us license to get deeper into the nitty gritty here, which rocks

Reply:
yep that was the idea!